### ISSUE TITLE
Rule against `await await` 

### ISSUE THREAD

**Issue by not-my-profile** [denoland/deno_lint#1168]
While the number of `await` keywords can affect which promise job is executed first (because [promise jobs have to be enqued even if the promise is fulfilled](https://tc39.es/ecma262/multipage/control-abstraction-objects.html#sec-performpromisethen)), in sane code you usually only ever want one `await` keyword for one expression. And multiple `await` keywords are usually just an accident, so it would be nice to have a rule to detect this.

E.g. Invalid:

```ts
await await fetch('/test');
```

Valid:

```ts
await fetch('/test');
```

**Comment by bartlomieju** [comment]
Sounds reasonable, PRs are welcome

**Comment by dsherret** [comment]
Has anyone ever written this kind of code though? Is a rule worth it considering the extra time necessary to check for this condition?

**Comment by dsherret** [comment]
It seems people write this kind of code: https://github.com/search?type=code&q=%22await+await+%22+language%3AJavaScript&l=JavaScript (maybe when coming from other languages like c# where this is necessary)

**Comment by not-my-profile** [comment]
I have written this code by accident ... you can easily end up with `await await` when copy'n'pasting code.

**Comment by dsherret** [comment]
Maybe deno fmt should just remove multiple awaits.

**Comment by not-my-profile** [comment]
As I explained in the issue description `await await` can result in different behavior than just `await`, so I think this change would be too dangerous for `deno fmt`, which I'd expect to **never** change the semantics of code.

**Comment by nayeemrmn** [comment]
I'm surprised the tsc warning isn't shown in this case
```
'await' has no effect on the type of this expression. deno-ts(80007)
```

> While the number of `await` keywords can affect which promise job is executed first

Does that really happen when `await`ing a non-promise?

**Comment by not-my-profile** [comment]
> I'm surprised the tsc warning isn't shown in this case

Oh, ... yeah me too. In case we get that warning to work then this lint is probably redundant.

> Does that really happen when awaiting a non-promise?

Yes.

**Comment by not-my-profile** [comment]
I guess we should open a new issue at `denoland/deno` about that tsc warning not being reported for `await await fetch('/test');`?

**Comment by dsherret** [comment]
> I guess we should open a new issue at denoland/deno about that tsc warning not being reported for await await fetch('/test');?

It is reported as a warning

![image](https://github.com/denoland/deno_lint/assets/1609021/3f32599b-d27c-4b79-b3c2-bf62acc92502)

**Comment by not-my-profile** [comment]
But not by `deno check`?

**Comment by nayeemrmn** [comment]
> ![image](https://user-images.githubusercontent.com/1609021/253674030-0d149183-23a0-4b5b-b1d8-69f700527840.png)

Ah, seems to work inconsistently:
![image](https://github.com/denoland/deno_lint/assets/29990554/ca6e9e01-292e-462e-a932-d74a0c09c752)

**Comment by dsherret** [comment]
Oh yeah. Looks like typescript is only reporting this as a suggestion diagnostic, which we don't surface. Perhaps it being reported via deno lint would still be good. I'm going to reopen.

**Comment by not-my-profile** [comment]
Slightly getting off-topic but would it make sense to add a `deno check` flag to also report suggestion diagnostics? Although it seems like the `tsc` CLI doesn't have such a flag.
